contracts/staking/PoSValidatorManager.sol

```diff
@@ -337,52 +325,36 @@ abstract contract PoSValidatorManager is IPoSValidatorManager, ValidatorManager
 
         // Ensure the delegator is active
         Delegator memory delegator = $._delegatorStakes[delegationID];
+        Validator memory validator = getValidator(validationID);
```

Comment:
If the validation period has already ended, there won't be any uptime message able to be provided in the call to `initializeEndDelegation`, and the uptime submitted when the validator was ended should be used to calculate the delegators rewards.